base/io.jl

```diff
@@ -1520,3 +1520,16 @@ function countlines(io::IO; eol::AbstractChar='\n')
 end
 
 countlines(f::AbstractString; eol::AbstractChar = '\n') = open(io->countlines(io, eol = eol), f)::Int
+
+# From Pkg.jl
+"""
+    UnstableIO <: IO
+
+Type unstable wrapper around an IO to avoid specializing write, get, and print
+"""
+struct UnstableIO <: IO
```

Comment:
This is probably beside the point, but this is insufficient to avoid inference inferring these functions on specialized IO. For that you'd need to add an extra inferencebarrier in the UnstableIO constructor.

Reply:
It worked pretty well in practice. :p

Reply:
@Keno, would that look like the following?

```julia
struct UnstableIO <: IO
    io::IO
    UnstableIO(@nospecialize(x)) = new(inferencebarrier(x))
end
```

Reply:
@Keno I added an `inferencebarrier` in [1f0876f](https://github.com/JuliaLang/julia/pull/52249/commits/1f0876fb5aec811335c7a8daee1a2eac34ca1441)

Reply:
This struct also doesn't seem compliant with the IO interface, so I am very reluctant to have this merged into Base. Perhaps make this `<: AbstractPipe` might suffice though, as then you get all of the required methods "for free"

Reply:
There's an `IO` interface? Let me see if I can find it. I'm going to see if  I can figure this out quickly. Otherwise, I will separate this into another pull request.